torch/csrc/jit/operator.h

```diff
@@ -0,0 +1,68 @@
+// in memory description of all ATen Ops similar to Caffe2 schema
+// once C10 exists this can be removed, or stubbed out, but we need
+// it now to implement correct semantic checking for script
+#pragma once
+#include "ATen/ATen.h"
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/function_schema.h"
+#include "torch/csrc/jit/stack_functions.h"
+
+namespace torch { namespace jit {
+
+FunctionSchema parseSchema(const std::string& decl);
+
+using OperationCreator = std::function<Operation(Node*)>;
+
+struct Operator {
+  Operator(FunctionSchema schema, OperationCreator op, OperationCreator op_const_attributes = nullptr)
+    : schema(std::move(schema))
+    , op(op)
+    , op_const_attributes(op_const_attributes) {}
+
+  Operator(const std::string& schema, OperationCreator op, OperationCreator op_const_attributes = nullptr)
+    : Operator(parseSchema(schema), op, op_const_attributes) {}
+
+  // primitive registration, creates a vararg schema with no required
+  // arguments
```

Comment:
this is confusing. what are we calling primitive operations now? almost none of them are varargs, why is that the default? do you mean things like Load? are the torch functions called builtin? we should really have a glossary somewhere

Reply:
Yeah, I will clarify the comment. Basically "primitive" is just suppose to mean: I want anything with this kind to match this implementation. It is used for things like `While`, `If`, and `Load` where the typesystem is not rich enough to capture the valid types, and where the node was only ever inserted by the compiler anyway so its schema is correct by construction.